### PR TITLE
updated gitignore docs for 2.3

### PR DESCRIPTION
**Pull request by sprankhub**
- updated GitHub link
- updated gitignore example by including 2.3 changes

## This PR is a:

- [ ] New topic
- [X] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged, it will include the latest 2.3 changes.

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-gitignore.html

### CODE REVIEW

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.3/config-guide/prod/config-reference-gitignore.md [hunk 1]
```diff
@@ -1 +1,179 @@
-../../../v2.2/config-guide/prod/config-reference-gitignore.md
```
Comment (jcalcaben) [inline guides/v2.3/config-guide/prod/config-reference-gitignore.md]: This is a symlink file. Replacing the contents of this file will break the build process.

To properly update this version of the file, please delete the symlink file and create a new file with the updated content.

Comment (sprankhub) [reply guides/v2.3/config-guide/prod/config-reference-gitignore.md]: Sorry, did not notice that. Fixed!

**Comment by jcalcaben** [comment]
running tests

**Pull request merged**